packages/perseus/src/error-boundary.tsx

```diff
@@ -22,13 +22,24 @@ class ErrorBoundary extends React.Component<Props, State> {
     componentDidCatch(error: Error, info: any) {
         this.setState({error: error.toString()});
         this.props.onError?.(error, info);
-        Log.error("Perseus error boundary caught error", Errors.Internal, {
-            cause: error,
-            loggedMetadata: {
-                componentStack: info.componentStack,
-                ...this.props.metadata,
+        Log.error(
+            // NOTE(jeremy): We concatenate the error messsage here. Typical
+            // Khan Academy error handling guidance says that you should never
+            // "build" the error message that might be sent to our error
+            // reporting tool (currently Sentry). However, if we don't
+            // differentiate between the different errors that are thrown, they
+            // all end up being grouped as a single Sentry event, which is very
+            // unhelpful.
+            "Unhandled Perseus error: " + error.message,
+            Errors.Internal,
+            {
+                cause: error,
+                loggedMetadata: {
+                    componentStack: info.componentStack,
```

Comment:
If `info` is null or undefined, this will throw. Maybe it's better to do `info?.componentStack`?

Reply:
The React [docs](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) seem to imply that it will always be provided.  But I'm happy to make this uber fault-tolerant. The last thing we want to do is cause some sort of recursive error scenario. :)

Reply:
Ok. I've made it more resilient if `info` is missing `componentStack`.  And added tests... because. :)